### PR TITLE
Move Cluster tests to t.Run

### PR DESCRIPTION
**Pull request by mcncl**
refactor(cluster resource tests): Migrate the Cluster resource tests to use the t.run approach

## PR checklist:
- [ ] `docs/` updated
- [x] tests added
- [ ] an example added to `examples/` (useful to demo new field/resource)
- [x] `CHANGELOG.md` updated with pending release information

### CODE REVIEW

File: buildkite/resource_cluster_test.go [hunk 1]
```diff
@@ -9,105 +9,130 @@ import (
...
+							resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", randNameUpdated)),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "id"),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "uuid"),
+						),
+					},
+				},
+			})
+		}
+		t.Run("with basic settings", func(t *testing.T) {
+			testCase(t, basic(randName), basic(randNameUpdated))
```
Comment (jradtilbrook) [inline buildkite/resource_cluster_test.go:98]: There isn't much point nesting this at this level with the `testCase` function. I think it should just merged above and run under the "updates a cluster" run

File: buildkite/resource_cluster_test.go [hunk 2]
```diff
@@ -9,105 +9,130 @@ import (
...
-					return c.UUID.ValueString(), nil
+	t.Run("Imports a Cluster", func(t *testing.T) {
+		randName := acctest.RandString(5)
+		check := resource.ComposeAggregateTestCheckFunc(
+			testAccCheckClusterExists("buildkite_cluster.foo", &c),
+			resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", randName)),
+		)
+
+		testCase := func(t *testing.T, config string) {
+			resource.Test(t, resource.TestCase{
```
Comment (jradtilbrook) [inline buildkite/resource_cluster_test.go:110]:
```suggestion
			resource.ParallelTest(t, resource.TestCase{
```

File: buildkite/resource_cluster_test.go [hunk 3]
```diff
@@ -9,105 +9,130 @@ import (
...
+				CheckDestroy:             testAccCheckClusterDestroy,
+				Steps: []resource.TestStep{
+					{
+						Config: config,
+						Check:  check,
+					},
+					{
+						ResourceName: "buildkite_cluster.foo",
+						ImportStateIdFunc: func(s *terraform.State) (string, error) {
+							return c.UUID.ValueString(), nil
```
Comment (jradtilbrook) [inline buildkite/resource_cluster_test.go:122]: Oh I didn't realise cluster is using UUID for importing. We'll need to change this for v1 to use ID like everything else

File: buildkite/resource_cluster_test.go [hunk 4]
```diff
@@ -9,105 +9,130 @@ import (
...
+							resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", randName)),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "id"),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "uuid"),
+						),
+					},
+					{
+						Config: newConfig,
+						Check: resource.ComposeAggregateTestCheckFunc(
+							testAccCheckClusterExists("buildkite_cluster.foo", &c),
+							testAccCheckClusterRemoteValues(&c, fmt.Sprintf("%s_test_cluster", randNameUpdated)),
```
Comment (jradtilbrook) [inline buildkite/resource_cluster_test.go:88]: This only checks the name but I think it should also check the other properties to make sure its all working correctly

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,7 @@ All notable changes to this project will be documented in this file.
 
 - Move archive pipeline config to provider [[PR #354](https://github.com/buildkite/terraform-provider-buildkite/pull/354)] @jradtilbrook
 - SUP-1076 Convert testing to framework [[PR #361](https://github.com/buildkite/terraform-provider-buildkite/pull/361)] @jradtilbrook
+    - SUP-1076 Move Cluster tests to use t.Run [[PR #365](https://github.com/buildkite/terraform-provider-buildkite/pull/365) @mcncl
```
Comment (jradtilbrook) [inline CHANGELOG.md:9]:
```suggestion
- SUP-1076 Move Cluster tests to use t.Run [[PR #365](https://github.com/buildkite/terraform-provider-buildkite/pull/365)] @mcncl
```

**Review by jradtilbrook — CHANGES_REQUESTED** [review]

File: buildkite/resource_cluster_test.go (hunk 1 above)
Comment (mcncl) [reply buildkite/resource_cluster_test.go:98]: @jradtilbrook there may be a time in which we want to add additional runs to this, such as using the `complex` config.

File: buildkite/resource_cluster_test.go (hunk 3 above)
Comment (mcncl) [reply buildkite/resource_cluster_test.go:122]: Agreed, but that shouldn't be done in this PR.

File: CHANGELOG.md (hunk 5 above)
Comment (mcncl) [reply CHANGELOG.md:9]: The above makes it a nested list, showing that it's a part of the same Linear work.

File: buildkite/resource_cluster_test.go (hunk 2 above)
Comment (mcncl) [reply buildkite/resource_cluster_test.go:110]: This errors if you use `Parallel` due to `foo` being used by both runs.

File: buildkite/resource_cluster_test.go [hunk 6]
```diff
@@ -9,105 +9,130 @@ import (
...
+		check := resource.ComposeAggregateTestCheckFunc(
+			testAccCheckClusterExists("buildkite_cluster.foo", &c),
+			testAccCheckClusterRemoteValues(&c, fmt.Sprintf("%s_test_cluster", randName)),
+			resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", randName)),
+			resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "id"),
+			resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "uuid"),
+		)
+
+		testCase := func(t *testing.T, config string) {
+			resource.Test(t, resource.TestCase{
```
[the remark this answers is not in the document]
Comment (mcncl) [reply buildkite/resource_cluster_test.go:44]: This errors if you use `Parallel` due to `foo` being used by both runs.

[the remark this answers is not in the document]
Comment (mcncl) [reply buildkite/resource_cluster_test.go:44]: This is why the tests now fail on `Import`.

[the remark this answers is not in the document]
Comment (mcncl) [reply buildkite/resource_cluster_test.go:44]: Latest run when only using one `t.Run` passes. Using `Parallel` with multiple `t.Run` which share the resource model will fail.

File: buildkite/resource_cluster_test.go (hunk 4 above)
Comment (mcncl) [reply buildkite/resource_cluster_test.go:89]: With the `basic` config, there is only the `name` set. It also checks that the ID and UUID are set.

Comment (mcncl) [reply buildkite/resource_cluster_test.go:89]: Added additional to ensure unchanged remain `null`

File: buildkite/resource_cluster_test.go (hunk 3 above)
Comment (jradtilbrook) [reply buildkite/resource_cluster_test.go:127]: No thats right. I added SUP-1380

File: CHANGELOG.md (hunk 5 above)
Comment (jradtilbrook) [reply CHANGELOG.md:9]: Oh I see 👍 just a missing bracket then

**Review by jradtilbrook — CHANGES_REQUESTED** [review]
I think the test structure needs to change so that we can run these tests in parallel. I'd say the double nesting of t.Run isn't helping in that case and a single level with a bit of duplication is an easier approach

**Comment by mcncl** [comment]
@jradtilbrook I'll change it. I do feel as though that's a preference thing, but if it's going to block the PR then it's not a hill I'm going to die on.

**Comment by jradtilbrook** [comment]
> @jradtilbrook I'll change it. I do feel as though that's a preference thing, but if it's going to block the PR then it's not a hill I'm going to die on.

Yeah because the tests are already slow, we should try to speed them up and parallelise them if we can

**Review by jradtilbrook — APPROVED** [review]

**Pull request merged**